static/scripts/chatLoggedin.js

```diff
@@ -1,8 +1,41 @@
+function extractRoomId(matrixUserId) {
+	if (!matrixUserId) {
+		return '';
+	}
+	const servername = matrixUserId.substr(matrixUserId.indexOf(':') + 1);
+
+	// extract room id from url
+	let roomId = null;
+	let roomType = null;
+	const { pathname } = window.location;
+	// > from course
+	const courseMatches = RegExp('/courses/([^/]+).*').exec(pathname);
+	if (courseMatches && courseMatches.length >= 2 && courseMatches[1] !== 'add') {
+		roomId = courseMatches[1];
+		roomType = 'course';
+	}
+	// > from team
+	const teamMatches = RegExp('/teams/([^/]+).*').exec(pathname);
+	if (teamMatches && teamMatches.length >= 2 && teamMatches[1] !== 'add') {
+		roomId = teamMatches[1];
+		roomType = 'team';
+	}
+
+	if (!roomId || !roomType || !servername) {
+		return '';
+	}
+
+	// build matrix room id
+	return `#${roomType}_${roomId}:${servername}`;
+}
+
 function loadChatClient(session = null) {
```

Comment:
The suggested changes do not work when `null` is passed to the function, because it can't read `userId` from null. Is there a similar format which allows to pass `null` instead of an object?

Reply:
argh, right...